selfdrive/ui/qt/widgets/assistant.cc

```diff
@@ -0,0 +1,94 @@
+#include "assistant.h"
+
+AssistantOverlay::AssistantOverlay(QWidget *parent) : QLabel(parent) {
+
+  setStyleSheet("QLabel {"
+                "  background-color: #373737;"
+                "  border-radius: 20px;"
+                "  font-family: 'Inter';"
+                "  font-size: 60px;"
+                "  color: white;"  // Text color
+                "}");
+
+  // Set up the animations
+  showAnimation = new QPropertyAnimation(this, "geometry");
+  showAnimation->setDuration(250); // Duration in milliseconds
+
+  hideAnimation = new QPropertyAnimation(this, "geometry");
+  hideAnimation->setDuration(250);
+
+  int height = 100; // Fixed height
+  setGeometry(0, 0, 0, height);
+
+  hideTimer = new QTimer(this);
+  connect(hideTimer, &QTimer::timeout, this, &AssistantOverlay::animateHide);
+
+  QObject::connect(uiState(), &UIState::uiUpdate, this, &AssistantOverlay::updateState);
+  hide();
+}
+
+void AssistantOverlay::animateShow() {
+  parentCenterX = parentWidget()->width() / 2;
+  finalWidth = parentWidget()->width() * 0.5;
+  startX = parentCenterX - finalWidth / 2;
+  QRect startRect(parentCenterX, 0, 0, height()); // Centered, zero width
+  QRect endRect(startX, 0, finalWidth, height()); // Adjusted x, final width
+  showAnimation->setStartValue(startRect);
+  showAnimation->setEndValue(endRect);
+  show();
+  showAnimation->start();
+}
+
+void AssistantOverlay::animateHide() {
+  parentCenterX = parentWidget()->width() / 2;
+  finalWidth = parentWidget()->width() * 0.5;
+  startX = parentCenterX - finalWidth / 2;
+  QRect startRect(startX, 0, finalWidth, height()); // Adjusted x, final width
+  QRect endRect(parentCenterX, 0, 0, height()); // Centered, zero width
+  hideAnimation->setStartValue(startRect);
+  hideAnimation->setEndValue(endRect);
+  hideAnimation->start();
+  hideTimer->stop();
+}
+
+void AssistantOverlay::updateText(QString text) {
+  this->setText(text);
+  this->setAlignment(QFontMetrics(this->font()).horizontalAdvance(text) > this->finalWidth ? Qt::AlignRight : Qt::AlignCenter);
+}
+
+void AssistantOverlay::updateState(const UIState &s) {
```

Comment:
simplify by using the error state instead of using isValid()? Need to make an integration test to send SpeechToText messages.